### PR TITLE
Fix squeeze loader bug and some other refinement

### PR DESCRIPTION
**Pull request by yangw1234**
## What changes were proposed in this pull request?

1. Fix squeeze loader bug and some other refinement
2. Fix saving tensorflow scalar variable error

## How was this patch tested?

added unit tests

### CODE REVIEW

**Comment by yangw1234** [comment]
pr validation pass

File: pyspark/bigdl/util/tf_utils.py [hunk 1]
```diff
@@ -117,9 +117,14 @@ def save_variable_bigdl(tensors, target_path, bigdl_type="float"):
     :param bigdl_type: model variable numeric type
     :return: nothing
     """
+    import numpy as np
     jtensors = {}
     for tn in tensors.keys():
-        jtensors[tn] = JTensor.from_ndarray(tensors[tn])
+        if not isinstance(tensors[tn], np.ndarray):
```
Comment (yiheng) [inline pyspark/bigdl/util/tf_utils.py]: when tensors[tn] will be an array?

Comment (yangw1234) [reply pyspark/bigdl/util/tf_utils.py]: when the variable is a scalar, tensors[tn] is not a `np.ndarray` but a single number. `JTensor.from_ndarray()` will fail on the single number case.

Comment (natnaelino) [reply pyspark/bigdl/util/tf_utils.py]: Same here fix single number fail np.ndarary ?

**Pull request merged**